libvast/vast/detail/legacy_deserialize.hpp

```diff
@@ -119,6 +119,22 @@ class legacy_deserializer { // an inspector for CAF inspect
     return apply_int(x);
   }
 
+#if VAST_CLANG
+  // For whatever reason, clang treats long and unsigned long as separate
+  // types from int64_t and uint64_t respectively, so we need to add another
+  // two overloads.
+
+  result_type apply(unsigned long& x) {
+    static_assert(sizeof(x) == sizeof(uint64_t));
+    return apply(reinterpret_cast<uint64_t&>(x));
```

Comment:
```suggestion
    return apply(static_cast<uint64_t&>(x));
```

Shouldn't this be a static cast?

Reply:
It can probably be one, yes.

I also still don't quite understand _why_ we need this, and I think we also may have to restrict the `#if` to `VAST_CLANG && VAST_MACOS`.

Reply:
I now compile with Clang 13, and it complains about "class member cannot be redeclared" so we need a more exact condition like `VAST_CLANG` && `VAST_MACOS`. The explanation of the problem can probably be inside stdint.h.

Reply:
Okay, I think we can consider this one resolved.